substrate/client/rpc-spec-v2/src/sudo_session_keys/mod.rs

```diff
@@ -0,0 +1,53 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Substrate sudo sessions key API.
+
+use serde::{Deserialize, Serialize};
+
+#[cfg(test)]
+mod tests;
+
+pub mod api;
+pub mod sudo_session_keys;
+
+/// The result of an RPC method.
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+#[serde(untagged)]
+pub enum MethodResult {
+	/// Method generated a result.
+	Ok(MethodResultOk),
+	/// Method ecountered an error.
+	Err(MethodResultErr),
```

Comment:
This is a little bit awkward because the spec could use the JSON-RPC error object but I suspect that some RPC client implementations may just throw away the error or something.

Reply:
Yep I also found that a bit weird:: https://github.com/paritytech/json-rpc-interface-spec/pull/104#issuecomment-1731291605

Reply:
Oki doki, we have brought inline the archive and sudo sessions keys. After all, we have discussed in the spec that this would be more appropriate for type script users